### PR TITLE
Added "x power y" in C

### PR DESCRIPTION
**Pull request by YashovardhanSiramdas**
**Changes:**
Added x power y in C. This Algorithm falls under Divide and Conquer.

### CODE REVIEW

File: code/divide_conquer/src/x_power_y/x_power_y.c [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    temp = power(x, y / 2);
+
+    if (y % 2 == 0)
+        return (temp*temp);
+
+    else {
+        if (y > 0)
+            return (x * temp * temp);
+        else
+            return ((temp * temp) / x);
```
Comment (arnavb) [inline code/divide_conquer/src/x_power_y/x_power_y.c]: No need for the nested parentheses.

File: code/divide_conquer/src/x_power_y/x_power_y.c [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ 
+float 
+power(float x, int y)
+{
+    float temp;
+
+    if (y == 0)
+       return (1);
+
+    temp = power(x, y / 2);
```
Comment (arnavb) [inline code/divide_conquer/src/x_power_y/x_power_y.c]: Remove float temp above, and put float here.

**Comment by YashovardhanSiramdas** [comment]
Made suggested changes.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**